Bluepill-cli/Bluepill-cli/Bluepill/Connections/BPTestBundleConnection.m

```diff
@@ -196,6 +202,30 @@ - (id)_XCT_testBundleReadyWithProtocolVersion:(NSNumber *)protocolVersion minimu
 
 - (id)_XCT_didBeginInitializingForUITesting {
     [BPUtils printInfo:INFO withString:@"Start initialization UI tests"];
+
+    self.uiTargetAppFifoFile = [[BPSimulatorFifoFile alloc]
+                                    initAtPath:[NSString stringWithFormat:@"/tmp/ui_target_app_output_%@", [self.simulator.device UDID]]
+                                    withDeviceDataPath:self.simulator.device.dataPath];
+
+    int fd = open([self.uiTargetAppFifoFile.absolutePath UTF8String], O_RDWR);
+    self.uiTargetOutputHandle = [[NSFileHandle alloc] initWithFileDescriptor:fd];
+
+    BPWriter *simulatorLogWriter = self.context.simulatorLogWriter;
+
+    NSRegularExpression *regexLastNewline = [NSRegularExpression regularExpressionWithPattern:@"\n\\Z" options:0 error:NULL];
+
+    self.uiTargetOutputHandle.readabilityHandler = ^(NSFileHandle *handle) {
```

Comment:
This is duplicate code for `stdOutHandle.readabilityHandler` in BPSimulator and we put a lot effort in BPTreeParser to make sure the data handling is correct.

Any reason besides extra prefix that you can't use `self.context.simulator.stdOutputHandler`?

Reply:
Yes, to do the prefixing. This can be refactored to remove duplication.

Reply:
I don't think the prefix [APP] is that useful.. logs from test runner will be prefixed with `t =  `.

Would be much simpler if we just put` kOptionsStderrKey: self.context.simulator.stdOutputHandler`.

Reply:
The test side can also output stuff, see my example in PR. The prefix makes it clear what is produced by the app.

Reply:
I understand, but it is clear that stuff prefixed with **XCTestOutputBarrier** will be from test case and the rest will be from the app.

I'm not comfortable of duplicating the logic to handle the data from the NSFileHandler, for example, there are cases that the data chunk is not a complete unicode char (first half in previous chunk, and second half in the next half). This would cause `encoding:NSUTF8StringEncoding ` to crash and we handled this in treeParser. So please just limit this PR to only reuse the handler and we can add support for prefix later.

Reply:
Messages from the test side are not always prefixed with `XCTestOutputBarrier`, see my example above, "message from UI test 2" is not prefixed and was produced by the test code.

If you want to add support for that later then that's fine.

Reply:
Right, but I don't want to risk the stability for an extra feature. We can figure out a way to refactor the code and add support for this later :)